R/ggpairs.R

```diff
@@ -781,10 +766,27 @@ ggpairs <- function(
     cardinality_threshold = 15,
     progress = NULL,
     proportions = NULL,
-    legends = stop("deprecated")) {
-  warn_deprecated(!missing(legends), "legends")
-  warn_if_args_exist(list(...))
-  stop_if_params_exist(params)
+    legends = deprecated()) {
+  if(lifecycle::is_present(legends)){
+    lifecycle::deprecate_warn(
+      when = "2.2.2",
+      what = "ggpairs(legends)",
+      details = "Ability to put legends in each plot will be dropped in next releases."
+    )
+  }
+  if(lifecycle::is_present(params)){
+    lifecycle::deprecate_warn(
+      when = "2.2.2",
+      what = "ggpairs(params)"
+    )
+  }
+  if(lifecycle::is_present(list(...))){
+    lifecycle::deprecate_warn(
+      when = "2.2.2",
+      what = "ggpairs(...)"
+    )
+  }
```

Comment:
Could we use `rlang::check_dots_empty()`?

Reply:
This would require rlang >= 1.0.0

Reply:
Yes, I guess we can, but I needed to tweak it a bit in order to not throw an error and keep the function without breaking changes for the users.

1fc3d59f6ae750e2d261308fb89068cbe911d9f7

Reply:
Thank you.

`lifecycle::is_present(list())` is always `TRUE`. Now we're able to check if `...` exists